soc: intel_adsp: ipc: check for pending ack

tmleman opened the pull request:
This patch extends the driver API with a function to check if we are waiting for an ACK from the host.

This change will allow to solve the problem that occurs during the power state transitions. Occasionally, the Application decides to enter the power gating state after sending an IPC message, before receiving an ACK from the HOST. This results in broken IPC communication when coming back to Idle state.

tmleman commented:
@dnikodem fyi

andyross commented on soc/xtensa/intel_adsp/common/include/intel_adsp_ipc.h:
```diff
@@ -174,4 +174,18 @@ bool intel_adsp_ipc_send_message_sync(const struct device *dev,
+{
+	struct intel_adsp_ipc_data *devdata = dev->data;
+
+	return devdata->tx_ack_pending;
```
This looks racy.  What does the caller do with this information?  The only safe thing to do is poll for completion, which the driver already supports, no?

tmleman replied:
Yes, it is intended for poll only and yes, driver checks this before sending new message. Thing I want to do here is to resolve a problem that we have in SOF. Because after message was send and if there is no other messages we are going to Idle thread. If PG is enabled, we will enter D0i3 state and it can happen before host will give ACK after receiving last message.

tmleman committed:
soc: intel_adsp: ipc: check for pending ack

This patch modifies intel_adsp_ipc_is_complete function so it don't
report that IPC is completed when we are still waiting for ACK from
HOST.

This change will allow to solve the problem that occurs during the power
state transitions. Occasionally, the Application decides to enter the
power gating state after sending an IPC message, before receiving an ACK
from the HOST. This results in broken IPC communication when coming back
to Idle state.

Signed-off-by: Tomasz Leman <tomasz.m.leman@intel.com>

tmleman commented:
**Update**: instead of adding a new function, I change the operation of the existing one.
